src/helpers.rs

```diff
@@ -0,0 +1,77 @@
+//! Helpers for the urlencoded module
+//!
+//! Provides one liners for the common usage of urlencoded
+
+use ::iron::prelude::*;
+use ::iron::status;
+use ::std::fmt::{Display,Formatter};
+use ::std::fmt::Error as FmtError;
+use ::std::error::Error as StdError;
+use super::{UrlEncodedBody,UrlEncodedQuery,QueryMap};
+
+/// Error returned when the requested parameter is missing
+#[derive(Debug, PartialEq)]
+pub struct MissingParamError {
+    name: String
+}
+
+impl StdError for MissingParamError {
+    fn description(&self) -> &str {
+        "Missing parameter"
+    }
+}
+
+impl Display for MissingParamError {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), FmtError> {
+        write!(f, "Missing parameter {}", self.name)
+    }
+}
+
+/// Returns the parameters hashmap constructed from the request body
+///
+/// # Examples
+/// ```
+/// fn request_handler(req: &mut Request) => IronResult<Response> {
+///     let params = try!(require_body_params(req));
+/// ```
+pub fn require_body_params(req: &mut Request) -> IronResult<QueryMap> {
+    req.get::<UrlEncodedBody>()
+        .map_err(|err| IronError::new(err, status::BadRequest))
+}
+
+/// Returns the parameters hashmap constructed from the request query
+///
+/// # Examples
+/// ```
+/// fn request_handler(req: &mut Request) => IronResult<Response> {
+///     let params = try!(require_query_params(req));
+/// 
+/// ```
+pub fn require_query_params(req: &mut Request) -> IronResult<QueryMap> {
+    req.get::<UrlEncodedQuery>()
+        .map_err(|err| IronError::new(err, status::BadRequest))
```

Comment:
Generally speaking I think it's wrong to do so - you want to specify an intelligent and semantic response when there is an error, not just toss 400 at everything.